### DISCUSSION TITLE
Markdown Metadata to Content

### DISCUSSION THREAD

**Discussion by ghost** [facebook/docusaurus#5621] [Q&A]
Hi,

## Challenge
In order to provide some details to auditors for us its necessary for some tools to document RACI roles and responsibilities.

I imagine something like:

documentation-of-x.mdx
```markdown
---
title: Tool X
raci:
  - creating-accounts:
       r: [personA, personB, personC]
       a: [personD]
       i: [group1]
---
```

Somehow then I would like to pickt that information up and publish them as own page/docs.

Like
baseUrl/docs/raci.html
```
|Tool|Task|R|A|C|I
|-|-|-|-|-|-
|Tool X|creating-accounts|personA, personB, person C|person D|-|group1
```

|Tool|Task|R|A|C|I
|-|-|-|-|-|-
|Tool X|creating-accounts|personA, personB, person C|person D|-|group1

> Why do you not just add it to each page as markdown?
You might ask. Because we need a central page with all RACI content to be audited, auditors wont search 30 places to get some scattered RACI matrix. Also, having it in one MD is not an option as people will forget to update it even more often than when its in the matching tool docs.

## Question
[lifecycle-apis#postbuildprops](https://docusaurus.io/docs/next/lifecycle-apis#postbuildprops) seems to me a bit late, there is no pre-build anyhow.
Is there some smart way to do that? Since my school year finished I could even think of developing a plugin somehow.

Option B is a script to create the page, but I'd rather sew it into docusaurus.

**Comment by slorber** [comment]
This is not documented and there's no official api for this yet, but you can extend the docs plugin to add an extra `/docs/raci` route for you.

Discussed in multiple places, like here: https://github.com/facebook/docusaurus/issues/4138 (and linked issues)

```js
const docsPluginExports = require('@docusaurus/plugin-content-docs');

const docsPlugin = docsPluginExports.default;

function docsPluginEnhanced(...pluginArgs) {
  const docsPluginInstance = docsPlugin(...pluginArgs);

  return {
    ...docsPluginInstance,
    contentLoaded: async function ({content,actions}) {
      // Create default docs routes
      await blogPluginInstance.contentLoaded({content,actions});
      
      // Create your own extra routes
      const currentVersion = content.loadedVersion[0] // Assume single version: use loop for multiple docs versions
      const raciData = currentVersion.docs.map(doc => ({id: doc.id, raci: doc.frontMatter.raci}));
      actions.addRoute({
        path: "/docs/raci"
        component: "@site/src/components/RaciPage.js",
        modules: {
          raci: await createData('raci.json',JSON.stringify(raciData, null, 2));
        }
      })
    },
  };
}

module.exports = {
  ...docsPluginExports,
  default: blogPluginEnhanced,
};

```

```jsx
export default RaciPage({raci}) {
  // render raci data
}
```

The API is not great and this looks like a workaround but we want to support this better in the future so don't be too afraid

Note: if you use multiple docs plugin instances, or want to add raci data to your blog, you won't be able to collect all those datas into a single page because each plugin is isolated: each plugin receive the data it loaded.

---

Another possibility is to have a pre-build step that extracts the raci fdata rom your frontmatter and creates a json file (with a node script), that you can import in `src/pages/raci.js`. The drawback is you have to run the step to update the page content (ie no hot reload, but it doesn't feel like a big deal for this usecase)
